python/tempo/tsdf.py

```diff
@@ -34,15 +34,38 @@ def __init__(self, df, ts_col="event_ts", partition_cols=None, sequence_col = No
 
     self.df = df
     self.sequence_col = '' if sequence_col is None else sequence_col
+
+    ## Add customized check for string type for the timestamp. If we see a string, we will proactively created a double version of the string timestamp for sorting purposes and rename to ts_col
+    if (df.schema[ts_col].dataType == "StringType"):
+        sample_ts = df.limit(1).collect()[0][0]
+        self.__validate_ts_string(sample_ts)
+        self.__add_double_ts().withColumnRenamed("double_ts", self.ts_col)
+
     """
     Make sure DF is ordered by its respective ts_col and partition columns.
     """
 
+
   ##
   ## Helper functions
   ##
 
-  def __validated_column(self,df,colname):
+  def __add_double_ts(self):
+      """ Add a double (epoch) version of the string timestamp out to nanos
+      """
+      self.df = self.df.withColumn("nanos", (f.when(f.col(self.ts_col).contains("."),
+                                     f.concat(f.lit("0."), f.split(f.col(self.ts_col), '\.')[1])).otherwise(0)).cast(
+          "double")).withColumn("double_ts", f.col(self.ts_col).cast("long") + f.col("nanos"))
```

Comment:
You cannot simply cast the string column ts_col to long. I believe the test code in the notebook had an extra step here to convert the part ahead of the decimal to a timestamp before converting it to long in the final calculation.

Reply:
Yes, this is a bug. It's been fixed to just compute epoch time and store that as the 'long' version.

Reply:
Nice, looks good.